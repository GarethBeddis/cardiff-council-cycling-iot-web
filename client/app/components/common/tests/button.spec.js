import React from 'react'
import { shallow } from 'enzyme'

import Button from '../button'

// Prepare a mock function
const mockOnClick = jest.fn()

describe('Button Component', () => {
  // Reset the Mock
  beforeEach(() => {
    mockOnClick.mockClear()
  })
  // General Tests
  it('Renders without crashing', () => {
    shallow(<Button text='Test Button' />)
  })
  it('OnClick event fires', () => {
    let wrapper = shallow(<Button text='Test Button' onClick={mockOnClick} />)
    wrapper.find('button.btn').first().simulate('click')
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
  // Snapshot Tests
  describe('Snapshots', () => {
    it('Standard button style matches Snapshot', () => {
      let wrapper = shallow(<Button text='Test Standard Button' />)
      expect(wrapper).toMatchSnapshot()
    })
    it('Danger button style matches Snapshot', () => {
      let wrapper = shallow(<Button text='Test Danger Button' danger />)
      expect(wrapper).toMatchSnapshot()
    })
    it('Custom class button matches Snapshot', () => {
      let wrapper = shallow(<Button className='test' text='Test Standard Button' />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
