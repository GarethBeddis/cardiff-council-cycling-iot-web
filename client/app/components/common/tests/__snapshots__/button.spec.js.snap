// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshots Custom class button matches Snapshot 1`] = `
<button
  className="btn test"
>
  Test Standard Button
</button>
`;

exports[`Button Component Snapshots Danger button style matches Snapshot 1`] = `
<button
  className="btn danger"
>
  Test Danger Button
</button>
`;

exports[`Button Component Snapshots Standard button style matches Snapshot 1`] = `
<button
  className="btn"
>
  Test Standard Button
</button>
`;
